mailserver/limiter_test.go

```diff
@@ -0,0 +1,93 @@
+package mailserver
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestIsAllowed(t *testing.T) {
+	peerID := "peerID"
+	testCases := []struct {
+		t               time.Duration
+		shouldBeAllowed bool
+		db              func() map[string]time.Time
+		errMsg          string
+		info            string
+	}{
+		{
+			t:               5 * time.Millisecond,
+			shouldBeAllowed: true,
+			db: func() map[string]time.Time {
+				return make(map[string]time.Time)
+			},
+			errMsg: "Expected limiter not to allow with empty db",
+			info:   "Expecting limiter.isAllowed to not allow with an empty db",
```

Comment:
It does not make sense because of `shouldBeAllowed: true`. So this condition is actually allowed.